qemu/tests/sev_hotplug_mem.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from virttest import error_context
+from virttest.utils_test.qemu import MemoryHotplugTest
+
+from avocado import TestError
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class MemoryHotplugSimple(MemoryHotplugTest):
+
+    def get_mem_by_name(self, vm, name):
+        """
+        Return memory object and pc-dimm devices by given name
+        """
+        dev_ids = map(lambda x: "-".join([x, name]), ["mem", "dimm"])
+        devices = filter(None, map(vm.devices.get_by_qid, dev_ids))
+        return [_[0] for _ in devices]
+
+    def unplug_memory(self, vm, target_mem):
+        """Unplug the target memory, if the memory not exists,
+           hotplug it, then unplug it
+        """
+        devs = self.get_mem_by_name(vm, target_mem)
+        if not devs and self.params.get("strict_check") != "yes":
+            self.hotplug_memory(vm, target_mem)
+            self.check_memory(vm)
+        return super(MemoryHotplugSimple, self).unplug_memory(vm, target_mem)
+
+    def check_memory(self, vm):
+        """
+        Check is guest memory is really match assigned to VM.
+
+        :param vm: VM object, get VM object from env if vm is None.
+        """
+        error_context.context("Verify guest memory size", LOG_JOB.info)
+        threshold = float(self.params.get("threshold", 0.10))
+        cmd = "grep 'MemTotal:' /proc/meminfo"
+        guest_mem_size = vm.get_memory_size(cmd)
+        vm_mem_size = super(MemoryHotplugSimple, self).get_vm_mem(vm)
+        sev_rom_size = int(self.params.get("sev_rom_size", 0))
+        error_context.context("%s" % vm_mem_size, LOG_JOB.info)
+        error_context.context("%s" % guest_mem_size, LOG_JOB.info)
+        if abs(guest_mem_size + sev_rom_size - vm_mem_size) > vm_mem_size * \
+                threshold:
+            msg = ("Assigned '%s MB' memory to '%s'"
+                   "but, '%s MB' memory detect by OS" %
+                   (vm_mem_size, vm.name, guest_mem_size))
+            raise TestError(msg)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu memory hotplug test:
+    1) Boot sev-es guest.
+    2) Hotplug/unplug  memory device
+    3) Check hotpluged memory detect in guest OS
+    4) Hotplug/unplug memory device
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Start sev memory hotplug test", test.log.info)
+    hotplug_test = MemoryHotplugSimple(test, params, env)
+    target_mems = params["target_mems"]
+    login_timeout = int(params.get("login_timeout", 360))
```

Comment:
login_timeout = params.get_numeric("login_timeout", 360)

Reply:
Thanks Xueqiang, update the patch